### PR TITLE
8292780: misc tests failed "assert(false) failed: graph should be schedulable"

### PR DESCRIPTION
**Pull request by rwestrel**
PhaseMacroExpand::generate_partial_inlining_block() adds
LoadVectorMasked nodes to the IR graph and then
LoadNode::split_through_phi() tries to split one of them through phi
but because that method ignores the mask input to that LoadNode (it
only knows about control, memory and address inputs) the resulting
graph is broken. Fix I propose is to skip
LoadNode::split_through_phi() for those LoadVector nodes that have
extra inputs.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8292780](https://bugs.openjdk.org/browse/JDK-8292780): misc tests failed "assert(false) failed: graph should be schedulable"

### Reviewers
 * [Dean Long](https://openjdk.org/census#dlong) (@dean-long - **Reviewer**) ⚠️ Review applies to [826cabe7](https://git.openjdk.org/jdk/pull/10410/files/826cabe7d092da027d9effd580409f17ad20dd61)
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to [826cabe7](https://git.openjdk.org/jdk/pull/10410/files/826cabe7d092da027d9effd580409f17ad20dd61)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10410/head:pull/10410` \
`$ git checkout pull/10410`

Update a local copy of the PR: \
`$ git checkout pull/10410` \
`$ git pull https://git.openjdk.org/jdk pull/10410/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10410`

View PR using the GUI difftool: \
`$ git pr show -t 10410`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10410.diff">https://git.openjdk.org/jdk/pull/10410.diff</a>

</details>

### CODE REVIEW

**Review by dean-long — APPROVED** [review]
It looks like a reasonable bailout.

**Comment by dean-long** [comment]
@jatin-bhateja, please review this.

File: src/hotspot/share/opto/memnode.cpp [hunk 1]
```diff
@@ -1485,7 +1485,12 @@ static bool stable_phi(PhiNode* phi, PhaseGVN *phase) {
 }
 //------------------------------split_through_phi------------------------------
 // Split instance or boxed field load through Phi.
-Node *LoadNode::split_through_phi(PhaseGVN *phase) {
+Node* LoadNode::split_through_phi(PhaseGVN *phase) {
```
Comment (chhagedorn) [inline src/hotspot/share/opto/memnode.cpp:1488]:
```suggestion
Node* LoadNode::split_through_phi(PhaseGVN* phase) {
```

**Review by chhagedorn — APPROVED** [review]
That looks reasonable!

**Comment by TobiHartmann** [comment]
So this is a regression from [JDK-8252848](https://bugs.openjdk.org/browse/JDK-8252848), correct?

**Comment by rwestrel** [comment]
> So this is a regression from [JDK-8252848](https://bugs.openjdk.org/browse/JDK-8252848), correct?

Yes.

**Review by TobiHartmann — APPROVED** [review]
Thanks, I updated the affected versions. The fix looks good to me.

**Comment by rwestrel** [comment]
> @jatin-bhateja, please review this.

Hi @dean-long I have 3 reviews and the fix is fairly straightforward. Can I integrate or should I wait for @jatin-bhateja ?

**Comment by dean-long** [comment]
> > @jatin-bhateja, please review this.
> 
> Hi @dean-long I have 3 reviews and the fix is fairly straightforward. Can I integrate or should I wait for @jatin-bhateja ?

I think it's fine to integrate this now because it fixes the problem, but I thought @jatin-bhateja might have an improvement to the fix that doesn't involve bailing out of the optimization.  Maybe file a follow-up RFE for that?

**Comment by rwestrel** [comment]
> I think it's fine to integrate this now because it fixes the problem, but I thought @jatin-bhateja might have an improvement to the fix that doesn't involve bailing out of the optimization. Maybe file a follow-up RFE for that?

I filed: https://bugs.openjdk.org/browse/JDK-8294750

**Comment by rwestrel** [comment]
Thanks @dean-long @TobiHartmann @chhagedorn  for the reviews.

**Pull request closed without merging**